cuckoofilter.go

```diff
@@ -72,7 +79,26 @@ func (cf *Filter) Insert(data []byte) bool {
 	if cf.insert(fp, i2) {
 		return true
 	}
-	return cf.reinsert(fp, randi(i1, i2))
+	return cf.reinsert(fp, cf.Coinflip(i1, i2))
+}
+
+// Using % isn't *perfectly* uniform, but it really only matters when n is a
+// significant fraction of the rng's range. It's more than good enough for our
+// purposes since n is on the order of 10^6 and our rng is 63 bits (10^19); this
+// means the bias is on the order of 10^-13. For our use case, that's well below
+// the noise floor.
+func (cf *Filter) Intn(n int) int {
+	// we need to make sure it's strictly positive, so mask off the sign bit
+	return int(cf.rng.Next()&0x7FFF_FFFF_FFFF_FFFF) % n
+}
+
+// Coinflip returns either i1 or i2 randomly by examining the least significant
+// bit of the RNG.
```

Comment:
Comment nit: 

Coinflip returns either i1 or i2 randomly with about equal chance.

The rest is an implementation detail.